1164/en.md

```diff
@@ -0,0 +1,130 @@
+# LOJ-1164: Horrible Queries
+
+## Prerequisite:
+1. Segment Tree
+2. Range Operation: Lazy Propagation
+
+## Problem at a glance:
+In this problem we are required to perform two kinds of operation.
+1. Update a range by a given `value`
+2. Find the total sum of given range
+
+## Approach
+Its clear that we can use `cumulative sum` to answer the second type of query. However, updating a range of values using this approach would have a time complexity of **O(n)**, which could lead to a **"Time Limit Exceeded" (TLE)** error for larger inputs. If you are already familiar with **segment tree** then you can start with segment tree. But unlike segment tree we are required to **update range** which may lead to **TLE** To solve this problem we can use a variation of **segment tree** called **lazy propagation** 
+
+## Lazy Propagation at a glance
+Lazy propagation is an optimization technique used with segment trees. If you are not familiar with segment trees, it is recommended to learn about them before proceeding.
+
+With lazy propagation, instead of updating a range instantly, we only mark the root node of that range for future updates. We maintain an extra array to keep track of which nodes need to be updated. Here's an illustration:
+
+<img src="lazyPropagationExample.jpg">
```

Comment:
Let's use the markdown format to insert an image.
```suggestion
![example image of lazy propagation](lazyPropagationExample.jpg)
```

Reply:
done